framework/auth/views.py

```diff
@@ -167,24 +156,34 @@ def forgot_password_post(auth, **kwargs):
         if user_obj:
             # check forgot_password rate limit
             if throttle_period_expired(user_obj.email_last_sent, settings.SEND_EMAIL_THROTTLE):
-                # new random verification key, allows OSF to check whether the reset_password request is valid,
-                # this verification key is used twice, one for GET reset_password and one for POST reset_password
-                # and it will be destroyed when POST reset_password succeeds
-                user_obj.verification_key = generate_verification_key()
-                user_obj.email_last_sent = datetime.datetime.utcnow()
-                user_obj.save()
-                reset_link = furl.urljoin(
-                    settings.DOMAIN,
-                    web_url_for(
-                        'reset_password_get',
-                        verification_key=user_obj.verification_key
+                    # check if user is an unclaimed contributor, if is, then send a new confirmation email to user
+                    # instead of sending user a reset password link
+                if (user_obj.is_invited and
```

Comment:
@mfraezz this part is added by @chennan47 as a quick fix for issue https://openscience.atlassian.net/browse/OSF-6673.